mono/mini/mini-arm.c

```diff
@@ -5055,16 +5058,18 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 				}
 
 				code = emit_big_add (code, ARMREG_IP, cfg->frame_reg, cfg->stack_usage + prev_sp_offset);
+				if (cfg->frame_reg == ARMREG_SP)
+					saved_ip_offset_fp = saved_ip_offset_sp;
 
 				/* Copy arguments on the stack to our argument area */
 				// FIXME a fixed size memcpy is desirable here,
 				// at least for larger values of stack_usage.
-				for (i = 0; i < call->stack_usage; i += sizeof (target_mgreg_t)) {
-					ARM_LDR_IMM (code, ARMREG_LR, ARMREG_SP, i + saved_ip_offset);
-					ARM_STR_IMM (code, ARMREG_LR, ARMREG_IP, i + saved_ip_offset);
+				for (i = 0; i < call->unaligned_stack_usage; i += sizeof (target_mgreg_t)) {
+					ARM_LDR_IMM (code, ARMREG_LR, ARMREG_SP, i + saved_ip_offset_sp);
+					ARM_STR_IMM (code, ARMREG_LR, ARMREG_IP, i + saved_ip_offset_fp);
```

Comment:
`s/saved_ip_offset_fp/saved_ip_offset_ip`?

Reply:
I know, it is confusing, but I meant that.
What to say that won't tell you what you already know?
There is the stack aka the frame. There are one or two pointers to it. The "floaty" one and "fixedy" one. Sometimes just one.
We are moving from the bottom of the floaty one, sp, to the top of the fixedy one, either sp or fp. As well, because we saved ip and were out of space, we pushed it, so we are adjusting for ip's saving..and the values are adjusting sp, and ip, ip as a temporary stand-in for fp, because, um, I'm not sure. Perhaps where it says ARMREG_IP it can actually say `cfg->frame_reg`, and then the entire push/pop and offseting can go away?

Let me think about that.

Reply:
Ok, I think the problem with the alternative..it will often work, but the problem is `emit_big_add`.
`i + saved_ip_offset_xx` cannot be unboundedly sized, and we are using ARMREG_IP to aborb that.

Reply:
Right. There is a 4k limit on parameter sizing:
```
mono_arch_tailcall_supported:
	// FIXME The limit here is that moving the parameters requires addressing the parameters
	// with 12bit (4K) immediate offsets. - 4 for TAILCALL_REG/MEMBASE
	res &= IS_SUPPORTED_TAILCALL (callee_info->stack_usage < (4096 - 4));
	res &= IS_SUPPORTED_TAILCALL (caller_info->stack_usage < (4096 - 4));
```

but what is being allowed here is huge amount of locals also.
If locals + params < 4k, then we could simplify, and that would be the overwhelming case.
It is ok asis, except for its error-proneness that caught me.

Reply:
Another plan is to store the parameters there in the first place.
Being sure not to use that space otherwise.
But I think this is ok.